templates/patterns/text_featured_media/pattern-text-featured-media.html.twig

```diff
@@ -0,0 +1,33 @@
+{#
+/**
+ * @file
+ * Text with featured media.
+ */
+#}
+
+{% set _column_css = ' ecl-col-md-6' %}
+
+{% if image.src is empty %}
+  {% set _column_css = '' %}
+{% endif %}
+
+{% if title %}
+  <h2 class="ecl-u-type-heading-2 ecl-u-mt-2xl ecl-u-mt-md-3xl ecl-u-mb-l">{{ title }}</h2>
+{% endif %}
+{% if image.src or text %}
+  <div class="ecl-row">
+    {% if image.src %}
+      <div class="ecl-col-12 ecl-col-md-6 ecl-u-order-md-last ecl-u-mb-l ecl-u-mb-md-none">
+        <figure class="ecl-media-container">
+          <img class="ecl-media-container__media" src="{{ image.src }}" alt="{{ image.alt }}">
+          {% if caption %}
+            <figcaption class="ecl-media-container__caption">{{ caption }}</figcaption>
+          {% endif %}
+        </figure>
+      </div>
+    {% endif %}
+    {% if text %}
+      <div class="ecl-col-12{{ _column_css }} ecl-editor">{{ text }}</div>
```

Comment:
Let's limit the use of extra variable to the minimum, a simple inline condition will do here.